src/subcommand.rs

```diff
@@ -85,11 +85,19 @@ impl Subcommand {
         )?;
         let root_dir = manifest.parent().unwrap();
 
-        let target_dir = target_dir.or_else(|| {
-            std::env::var_os("CARGO_BUILD_TARGET_DIR")
-                .or_else(|| std::env::var_os("CARGO_TARGET_DIR"))
-                .map(|os_str| os_str.into())
-        });
+        let target_dir = target_dir
+            .or_else(|| {
+                std::env::var_os("CARGO_BUILD_TARGET_DIR")
+                    .or_else(|| std::env::var_os("CARGO_TARGET_DIR"))
+                    .map(|os_str| os_str.into())
+            })
+            .map(|target_dir| {
+                if target_dir.is_relative() {
+                    std::env::current_dir().unwrap().join(target_dir)
+                } else {
+                    target_dir
+                }
```

Comment:
isn't this exactly what canonicalize does? turns a relative path into an absolute one and follows symlinks?

Reply:
It asks the os to perform that, yes. Except that this bit of code doesn't perform any lookups for the path to eg. resolve symlinks, which is impossible because the `target` folder doesn't exist when first calling this in a repository. With #4 merged `cargo apk` now fails in a fresh (or cleaned) crate.

Before you ask, I don't think we can call `canonicalize` on `.parent()` either because `CARGO_(BUILD_)TARGET_DIR` might be arbitrarily deep.

I assume you ask this because this bit of code is rather obnoxious and seems there should be a generic helper for in the standard library... concatenating some relative path to the current directory without checking if any of those directories exist doesn't sound all too uncommon to me.

Reply:
How about creating the target dir if it doesn't exist and then canonicalized? But I guess this works too.

Reply:
Yeah, that should do the trick as well. What do you prefer?

Reply:
I think we can make sure the target dir exists and return a canonical path to it.

Reply:
@dvc94ch We might also return the absolute or relative path and let the calling application (`cargo-apk`) deal with it?